lib/ansible/modules/cloud/amazon/code_build.py

```diff
@@ -0,0 +1,338 @@
+        found_last_modified = found['lastModified']
+        resp = update_project(client=client, params=formatted_params, module=module)
+        update_last_modified = resp['project']['lastModified']
+
+        if update_last_modified > found_last_modified:
+            resp['updated'] = True
+        return resp
+    # Or create new project:
+    try:
+        resp = client.create_project(**formatted_params)
+        return resp
+    except botocore.exceptions.ClientError as e:
+        module.fail_json(msg="Unable create project {0}: {1}".format(name, to_native(e)),
+                         exception=traceback.format_exc(), **camel_dict_to_snake_dict(e.response))
+    except botocore.exceptions.BotoCoreError as e:
+        module.fail_json(msg="Unable to create project {0}: {1}".format(name, to_native(e)),
+                         exception=traceback.format_exc())
+
+
+def update_project(client, params, module):
+    name = params['name']
+
+    try:
+        resp = client.update_project(**params)
+        return resp
+    except botocore.exceptions.ClientError as e:
+        module.fail_json(msg="Unable update project {0}: {1}".format(name, to_native(e)),
+                         exception=traceback.format_exc(), **camel_dict_to_snake_dict(e.response))
+    except botocore.exceptions.BotoCoreError as e:
+        module.fail_json(msg="Unable to update project {0}: {1}".format(name, to_native(e)),
+                         exception=traceback.format_exc())
+
+
+def delete_project(client, name, module):
+    try:
+        resp = client.delete_project(name=name)
+        return resp
+    except botocore.exceptions.ClientError as e:
+        module.fail_json(msg="Unable delete project {0}: {1}".format(name, to_native(e)),
+                         exception=traceback.format_exc(), **camel_dict_to_snake_dict(e.response))
+    except botocore.exceptions.BotoCoreError as e:
+        module.fail_json(msg="Unable to delete project {0}: {1}".format(name, to_native(e)),
+                         exception=traceback.format_exc())
+
+
+def describe_project(client, name, module):
+    project = {}
+    try:
+        projects = client.batch_get_projects(names=[name])['projects']
+        if len(projects) > 0 and isinstance(projects[0], dict):
+            project = projects[0]
+        return project
+    except botocore.exceptions.ClientError as e:
+        return project
+    except botocore.exceptions.BotoCoreError as e:
+        module.fail_json(msg="Error when calling client.batch_get_projects {0}: {1}".format(name, to_native(e)),
+                         exception=traceback.format_exc())
+
+
+def main():
+    argument_spec = ec2_argument_spec()
+    argument_spec.update(dict(
+        name=dict(required=True, type='str'),
+        description=dict(required=False, type='str'),
+        source=dict(required=True, type='dict'),
+        artifacts=dict(required=True, type='dict'),
+        cache=dict(required=False, type='dict'),
```

Comment:
required=False is the default so it can also be omitted. Same for any of the following.